cmd/build/v2/servicecontext.go

```diff
@@ -0,0 +1,38 @@
+package v2
+
+import (
+	"github.com/okteto/okteto/pkg/filesystem"
+	oktetoLog "github.com/okteto/okteto/pkg/log"
+	"github.com/okteto/okteto/pkg/model"
+	"github.com/okteto/okteto/pkg/repository"
+)
+
+type serviceContextInterface interface {
+	IsCleanBuildContext() bool
+}
+
+type serviceContext struct {
+	isClean bool
+}
+
+func (sc *serviceContext) IsCleanBuildContext() bool {
+	return sc.isClean
+}
+
+func getServiceContext(buildInfo *model.BuildInfo) *serviceContext {
+	wdCtrl := filesystem.NewOsWorkingDirectoryCtrl()
+	wd, err := wdCtrl.Get()
+	if err != nil {
+		oktetoLog.Infof("could not get working dir: %w", err)
+	}
+	gitRepo := repository.NewRepository(wd)
```

Comment:
Same, should we inject the repository from the build and not creating it again? It will let us test with integration tests in the future

Reply:
c4ade2c07260cf3dd979a7e0c69bc6bcda87c71d